server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -50,41 +50,58 @@ def _get_default_config():
             "diffexp_lfc_cutoff": 0.01,
         }
 
-    def _alias_annotation_names(self, axis, name):
+    @staticmethod
+    def _create_unique_column_name(df, prefix):
```

Comment:
Can we change this method's signature: `(name, columns)` 
1. Since you are only using `df.columns` no need to pass the whole dataframe, doing that makes knowing what the function does with the dataframe harder to understand.
2. `prefix` was not what I expected it to be. I thought it was going to be what we prefix non-unique column names with.

Reply:
will make change (1), and will add better comments and variable names for prefix, in an attempt to add clarity....